src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Serialization/ObjectReader.cs

```diff
@@ -182,13 +182,13 @@ public object ReadValue()
             case TypeCode.Null: return null;
             case TypeCode.Boolean_True: return true;
             case TypeCode.Boolean_False: return false;
-            case TypeCode.Int8: return _reader.ReadSByte();
```

Comment:
direct access of the reader makes the code harder to update since it's the reader itself that is being swapped out in the new world.